test/unit/gateways/blue_snap_test.rb

```diff
@@ -938,6 +958,55 @@ def forbidden_response
     MockResponse.new(403, '<xml>You are not authorized to perform this request due to inappropriate role permissions.</xml>')
   end
 
+  def fraudulent_purchase_response
+    body = <<~XML
+      <?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
+        <messages xmlns=\"http://ws.plimus.com\">
+          <message>
+            <error-name>FRAUD_DETECTED</error-name>
+            <code>15011</code>
+            <description>The request cannot be fulfilled for the current shopper. Please contact BlueSnap support for further details.</description>
+            <fraud-events>
+              <fraud-reference-id>6270209</fraud-reference-id>
+              <fraud-event>
+                <fraud-event-code>orderTotalDecline</fraud-event-code>
+                <fraud-event-decision>D</fraud-event-decision>
+                <fraud-event-expression>3729 &gt; 3728</fraud-event-expression>
+              </fraud-event>
+            </fraud-events>
+          </message>
+        </messages>
+    XML
+    MockResponse.new(400, body)
```

Comment:
Do you know why [the `MockResponse`](https://github.com/activemerchant/active_merchant/blob/6b8d81cdb8d2d886ad97ef890221ad6530642507/test/test_helper.rb#L355) is needed here? I ran the tests without it, and it's obviously needed for this gateway.
I worked on CyberSource recently, and it didn't use the `MockResponse` in the example responses, which is why I'm curious if you know why it's needed here: https://github.com/activemerchant/active_merchant/blob/master/test/unit/gateways/cyber_source_test.rb#L778

Reply:
I don't, and I don't remember having seen it before. I'm also curious, though. Maybe we should make a list of Unsolved Code Mysteries?

Reply:
I'm seeing us hosting a Spreedly episode of The Twilight Zone! I feel like that could become a very long list. 👻